### ISSUE TITLE
Add-on is broken on current Firefox for Android 50 beta (and aurora)

### ISSUE THREAD

**Issue by m-khvoinitsky** [m-khvoinitsky/dark-background-light-text-extension#46]
require() inside process-script causes:
`resource://gre/modules/commonjs/toolkit/loader.js:244`
`Component returned failure code: 0x80004002 (NS_NOINTERFACE) [nsISupports.QueryInterface]`
```
getZipFileContents@resource://gre/modules/commonjs/toolkit/loader.js:244:16
urlCache.zipContentsCache<@resource://gre/modules/commonjs/toolkit/loader.js:268:14
get@resource://gre/modules/commonjs/toolkit/loader.js:221:21
exists@resource://gre/modules/commonjs/toolkit/loader.js:330:30
resolveAsFile@resource://gre/modules/commonjs/toolkit/loader.js:596:7
resolveRelative@resource://gre/modules/commonjs/toolkit/loader.js:627:23
nodeResolve@resource://gre/modules/commonjs/toolkit/loader.js:673:23
@resource://gre/modules/commonjs/toolkit/loader.js:989:25
getRequirements@resource://gre/modules/commonjs/toolkit/loader.js:865:23
_require@resource://gre/modules/commonjs/toolkit/loader.js:761:32
require@resource://gre/modules/commonjs/toolkit/loader.js:757:12
make/require@resource://gre/modules/commonjs/toolkit/require.js:13:8
@chrome://dark-background-light-text/content/process-script.js:7:18
exports.main@resource://gre/modules/commonjs/toolkit/loader.js ->
 resource://jid1-qofqdk4qzufgwq-at-jetpack/lib/index.js:435:5
run@resource://gre/modules/commonjs/toolkit/loader.js ->
 resource://gre/modules/commonjs/sdk/addon/runner.js:23:64
startup/</<@resource://gre/modules/commonjs/toolkit/loader.js ->
 resource://gre/modules/commonjs/sdk/addon/runner.js:7:91
Handler.prototype.process@resource://gre/modules/Promise.jsm ->
 resource://gre/modules/Promise-backend.js:103:82
this.PromiseWalker.walkerLoop@resource://gre/modules/Promise.jsm ->
 resource://gre/modules/Promise-backend.js:96:31
this.PromiseWalker.scheduleWalkerLoop/<@resource://gre/modules/Promise.jsm ->
 resource://gre/modules/Promise-backend.js:83:94
```

Problem code in `resource://gre/modules/commonjs/toolkit/loader.js`:
```
  getZipFileContents(uri, baseURL) {
    // Make sure the path has a trailing slash, and strip off the leading
    // slash, so that we can easily check whether it is a path prefix.
    let basePath = addTrailingSlash(uri.JAREntry).slice(1);
 >>>let file = uri.JARFile.QueryInterface(Ci.nsIFileURL).file;

    let enumerator = zipCache.getZip(file).findEntries("(*.js|*.json|*/)");

    let results = new Set();
    for (let entry of XPCOMUtils.IterStringEnumerator(enumerator)) {
      if (entry.startsWith(basePath)) {
        let path = entry.slice(basePath.length);

        results.add(baseURL + path);
      }
    }

    return results;
  },
```

While desktop developer edition version (with e10s enabled) only warns: `Use of nsIFile in content process is deprecated.` but still works.

Nothing about this was mentioned [here - Add-on Compatibility for Firefox 50](https://blog.mozilla.org/addons/2016/09/01/compatibility-for-firefox-50/)

https://bugzil.la/1315547

**Comment by the8472** [comment]
This is unlikely to solve the issue, but you may be able to simplify your code by using the sdk's [remote/parent](https://developer.mozilla.org/en-US/Add-ons/SDK/Low-Level_APIs/remote_parent#Loading_modules_into_the_child_process) module which provide functionality equivalent to process scripts.

**Comment by m-khvoinitsky** [comment]
@the8472 thank you! For some reason I've missed this SDK module but it seems to be exactly what I need. I've just tested and the bug is not reproduced with remote/parent and, by the way, moving to remote/parent will also fix #36. Porting will not take much effort.
